apps/document-check-function/check-new-document/index.js

```diff
@@ -0,0 +1,32 @@
+import { sendDocumentStateAction } from './back-office-api-client.js';
+import { scanStream } from './clam-av-client.js';
+
+/**
+ * @param {boolean} isInfected
+ * @returns {string}
+ */
+const mapIsInfectedToMachineAction = (isInfected) => {
+	return isInfected ? 'check_fail' : 'check_pass';
+};
+
+/**
+ * @param {string} blobUri
+ * @returns {{caseId: string, guid: string}}
+ */
+const getBlobCaseIdAndGuid = (blobUri) => {
+	const uriParts = blobUri.split('/');
+
+	return { caseId: uriParts[1], guid: uriParts[2] };
+};
+
+/**
+ * @param {{bindingData: {uri: string}}} context
+ * @param {import('node:stream').Readable} myBlob
+ */
+export const checkMyBlob = async (context, myBlob) => {
+	const isInfected = await scanStream(myBlob);
+	const machineAction = mapIsInfectedToMachineAction(isInfected);
+	const { caseId, guid } = getBlobCaseIdAndGuid(context.bindingData.uri);
+
+	await sendDocumentStateAction(guid, caseId, machineAction);
```

Comment:
As part of SEC-04, we need to 'quarantine' documents which have been deemed unsafe. I think this is open to interpretation, because to some extent these files are 'quarantined' when they're first uploaded (in the sense that only this function can access them). It might be worth thinking about moving the file to a separate container where nothing has READ access and this function only has WRITE access.

Reply:
We're also going to need to raise some alarm here, for now it might just be worth adding in a console.warning so that at a minimum we can subscribe to the log group for warning messages.